src/app/packages/timeline/services/timeline.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { TestBed, inject } from '@angular/core/testing';
+
+import { TimelineService } from './timeline.service';
+
+import { HttpModule,XHRBackend,Response,ResponseOptions,Http } from '@angular/http';
+
+import { MockBackend } from '@angular/http/testing';
+
+import { Observable ,Observer} from 'rxjs';
+
+
+describe('TimelineService', () => {
+  let timeLineService,mockBackend,lastConnection,http ;
+
+  let response = 	{ data : [ 
+			{key: "a",value: 1},
+			{key: "b",value: 2} 
+			]};
+
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+		  providers: [
+		  	TimelineService,
+		  	{ provide: XHRBackend, useClass: MockBackend }
+		  ],
+		  imports:[HttpModule]
+		});
+	});
+
+  	beforeEach( inject([TimelineService,XHRBackend,Http], (_timeLineService,_mockBackend,_http) => {
+  		timeLineService = _timeLineService;
+  		mockBackend = _mockBackend;
+  		http = _http;
+
+  		mockBackend.connections.subscribe( (connection:any) => {
+    		if(connection.request.url == "//localhost:8037/api/mock/eventDrops/data") { 		
+	      		connection.mockRespond(new Response( new ResponseOptions({
+	      			body: JSON.stringify(response)
+	      		})));
+      		}
+
+      		if(connection.request.url == "error") {
+      			console.log('errpr'); 		
+	      		connection.mockError(new Error('Username or password is incorrect'));
+      		}
+    	});
+
+ 	}));
+
+	it('check all dependencies are defined properly', () => {
+		expect(timeLineService).toBeTruthy();
+		
+	});
+
+	it('check the method fetchData with mock data', () => {     
+ 		
+ 		timeLineService.fetchData().subscribe(result => {     
+     		expect(result.data.length).toBe(2);
+     	})
+ 		
+		timeLineService.fetchData('tmp').subscribe(result => {     
```

Comment:
I removed the check. The reason it passed by and didn't throw any exception is that the process never reached the subscribed function since the mock didn't called .next()